src/gitProviders/github.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from "react";
+import axios from 'axios';
+
+export class GitHub extends React.Component<any, any>{
+
+    constructor(props) {
+        super(props);
+        console.log(this.context);
+        this.handleChange = this.handleChange.bind(this);
+        this.createNewBranch = this.createNewBranch.bind(this);
+    }
+
+    async componentDidMount() {
+        await this.loadBranches();
+    }
+
+    private handleChange(event) {
+        const { target: { name, value } } = event;
+        this.props.onElementChange(name,value);
+        console.debug("Update state property: ", name, value);
+    }
+
+    private async loadBranches() {
+        const branches: Array<any> = await this.fetchBranches();
+        console.log("Branches:"+branches);
+        const branchOptions = branches.map((b) => {
+            let rObj = {};
+            rObj['value'] = b;
+            rObj['label'] = b;
+            return rObj;
+        });
+        this.props.onElementChange('branch',this.props.context.branch ? this.props.context.branch : branches[0]);
+        this.props.onElementChange('branchOptions',branchOptions);
+    }
+
+    const // @ts-ignore
```

Comment:
I'm not a real typescript pro - what does this empty declaration do?

Reply:
Nothing, that was some orphaned code.  I took care of it in the last commit.